### DISCUSSION TITLE
How to get the actual pixel size of a transformed sprite?

### DISCUSSION THREAD

**Discussion by awhillas** [bevyengine/bevy#8919] [Q&A] [answered]
This seems like it should be easy but I haven't found a straight answer on how to do this anywhere.
So I'm trying to have sprites spawn, move around and detect clicks on them and then do something. So in order to figure out if a sprite I clicked/touched I need to know its bounding box?

So I spawn new `SpriteBundle`s and apply a transform to them to scale them down (might scale them dynamically in the future)
```rust
...
            commands. Spawn((
                SpriteBundle {
                    transform: Transform::from_xyz(x, y, 0.0).with_scale(Vec3::splat(0.1)),
                    texture: asset_server.load("images/S.png"),
                    ..default()
                },
                ...
            ));
...
```
I found [this example](https://pastebin.com/4CejRLUS) of reading an images dimensions
```rust
fn confine(
    mut confinable_query: Query<(&mut Transform, &Handle<Image>), With<Confinable>>,
    assets: Res<Assets<Image>>
) {
    let window = window_query.get_single().unwrap();
 
    for (mut transform, image_handle) in confinable_query.iter_mut() {
        let image_dimensions = assets.get(image_handle).unwrap().size();
...
```
but it seems the `image_dimensions` are the original image (they don't change with or without scale transform applied) not the new scaled size? How do i get the bounding box of a scaled sprite in real time?

**Comment by awhillas** [comment, marked as answer]
[from](https://www.reddit.com/r/rust_gamedev/comments/14fkzwk/comment/jpfvdpq/?utm_source=share&utm_medium=web2x&context=3) but the gist is you can pull the scaling factor from the transform.
```rust
fn print_sprite_bounding_boxes(
    mut sprite_query: Query<(&Transform, &Handle<Image>), With<Sprite>>,
    assets: Res<Assets<Image>>,
) {
    for (transform, image_handle) in sprite_query.iter_mut() {
        let image_dimensions = assets.get(image_handle).unwrap().size();
        let scaled_image_dimension = image_dimensions * transform.scale.truncate();
        let bounding_box = Rect::from_center_size(transform.translation.truncate(), scaled_image_dimension);

        println!("bounding_box: {:?}", bounding_box);
    }
}
```
I guess Reddit wins between [Stack Overflow](https://stackoverflow.com/questions/76527423/how-to-get-the-actual-pixel-size-of-a-transformed-sprite-in-bevy) and here.

**Comment by vim-zz** [comment]
It's very strange i tried to pull out the example code into a running demo and I get the text at the center... what even more strange the font size is different here (i changed the text to 12345 to see that its in the center).

Now I am trying to understand what cause this...

<img width="374" alt="Screenshot 2024-01-21 at 21 28 10" src="https://github.com/bevyengine/bevy/assets/50141/75e0135a-3ed2-432c-b874-1e400ab55fcd">
